mutual-information-feature-selection/data/format_data.py

```diff
@@ -0,0 +1,75 @@
+# reformat the data
+import os
+import re
+
+import numpy as np
+import pandas as pd
+
+# make this runnable from other locations
+my_loc = os.path.dirname(os.path.abspath(__file__))
+os.chdir(my_loc)
+
+
+dataset = pd.read_csv('titanic.csv')
+
+
+# statistics about name
+
+def get_title(name):
+    title_search = re.search(r" ([A-Za-z]+)\.", name)
+    # If the title exists, extract and return it.
+    if title_search:
+
+        title = title_search.group(1)
+
+        # do some grouping
+        if title in ('Miss', 'Ms', 'Mlle'):
+            return 'miss'
+        if title in ('Mrs', 'Mme'):
+            return 'mrs'
+        if title in ['Lady', 'Countess', 'Capt', 'Col', 'Don', 'Dr', 'Major',
+                     'Rev', 'Sir', 'Jonkheer', 'Dona']:
+            return 'rare'
+
+        return title.lower()
+    return ""
+
+
+dataset['title'] = dataset['name'].apply(get_title)
+
+for title in dataset['title'].unique():
+    dataset[title] = dataset['title'].apply(lambda t: t == title)
+
+# alone or not
+dataset['alone'] = (dataset["sibsp"] + dataset["parch"]).apply(lambda x: x == 0)
+
+# in a cabin or not
+dataset['cabin'] = dataset['cabin'].apply(lambda c: not pd.isna(c))
+
+# sex to binary
+dataset['sex'] = dataset['sex'].apply(lambda s: s == 'female')*1
+
+# embarked port
+dataset['embarked port S'] = dataset['embarked'].apply(lambda x: x == 'S')
+dataset['embarked port C'] = dataset['embarked'].apply(lambda x: x == 'C')
+dataset['embarked port Q'] = dataset['embarked'].apply(lambda x: x == 'Q')
+
+# drop the folks with no age or fare
+dataset = dataset[np.isfinite(dataset['age'])]
+dataset = dataset[np.isfinite(dataset['fare'])]
+
+# make age discrete by grouping into 20s (rounding down)
+dataset['age'] = dataset['age'].apply(lambda a: 2*int(a // 20))
```

Comment:
I know that the math will produce different solutions. I just don't see the point of multiplying by 2. If we were grouping with 20s, why can't we simply do `int(a // 20)` and be happy with 1, 1, and 2 (as with Pau's example)? Otherwise, just group by tens with `int(a // 10)`.

The multiply by two just irks me. I wish there was an explanation for the multiply by 2 in the comment.